ome/datasource_baseline_compliance_report.go

```diff
@@ -5,111 +5,115 @@ import (
 	"terraform-provider-ome/clients"
 	"terraform-provider-ome/models"
 
-	"github.com/hashicorp/terraform-plugin-framework/diag"
-	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
 	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-type configurationReportDataSourceType struct{}
+var (
+	_ datasource.DataSource              = &configurationReportDataSource{}
+	_ datasource.DataSourceWithConfigure = &configurationReportDataSource{}
+)
+
+// NewConfigurationReportDataSource is a new datasource for configuration report
+func NewConfigurationReportDataSource() datasource.DataSource {
+	return &configurationReportDataSource{}
+}
+
+type configurationReportDataSource struct {
+	p *omeProvider
+}
 
-func (t configurationReportDataSourceType) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
-	return tfsdk.Schema{
+// Configure implements datasource.DataSourceWithConfigure
+func (g *configurationReportDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
```

Comment:
Pls add an error here

Reply:
applicable to all such instances

Reply:
Will be adding errors for the hard exit into the terraform diagnostics.

Reply:
resolved, we have added the error for data source and resource configure method

Reply:
This is causing errors in the provider lifecycle, so we're not adding these errors to configure method.